encoding/fuzz_test.go

```diff
@@ -0,0 +1,183 @@
+//go:build go1.18
+// +build go1.18
+
+package encoding_test
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/segmentio/parquet-go/encoding"
+	"github.com/segmentio/parquet-go/encoding/bytestreamsplit"
+	"github.com/segmentio/parquet-go/encoding/delta"
+	"github.com/segmentio/parquet-go/encoding/plain"
+	"github.com/segmentio/parquet-go/encoding/rle"
+	"github.com/segmentio/parquet-go/format"
+)
+
+func fuzzEncoding(fuzz func(e encoding.Encoding)) {
+	for _, test := range [...]struct {
+		scenario string
+		encoding encoding.Encoding
+	}{
+		{
+			scenario: "PLAIN",
+			encoding: new(plain.Encoding),
+		},
+
+		{
+			scenario: "RLE",
+			encoding: new(rle.Encoding),
+		},
+
+		{
+			scenario: "PLAIN_DICTIONARY",
+			encoding: new(plain.DictionaryEncoding),
+		},
+
+		{
+			scenario: "RLE_DICTIONARY",
+			encoding: new(rle.DictionaryEncoding),
+		},
+
+		{
+			scenario: "DELTA_BINARY_PACKED",
+			encoding: new(delta.BinaryPackedEncoding),
+		},
+
+		{
+			scenario: "DELTA_LENGTH_BYTE_ARRAY",
+			encoding: new(delta.LengthByteArrayEncoding),
+		},
+
+		{
+			scenario: "DELTA_BYTE_ARRAY",
+			encoding: new(delta.ByteArrayEncoding),
+		},
+
+		{
+			scenario: "BYTE_STREAM_SPLIT",
+			encoding: new(bytestreamsplit.Encoding),
+		},
+	} {
+		fuzz(test.encoding)
+	}
+}
+
+func FuzzByteArrayEncoding(f *testing.F) {
+	f.Fuzz(func(t *testing.T, input []byte) {
+		fuzzEncoding(func(e encoding.Encoding) {
+			fuzzByteArrayEncoding(t, input, e)
+		})
+	})
+}
+
+func FuzzFixedLenByteArrayEncoding(f *testing.F) {
+	f.Fuzz(func(t *testing.T, size int, input []byte) {
+		fuzzEncoding(func(e encoding.Encoding) {
+			fuzzFixedLenByteArrayEncoding(t, size, input, e)
+		})
+	})
+}
```

Comment:
Some encodings don't support byte arrays, do we need to add fuzzing tests for other data types?